### DISCUSSION TITLE
Clarification of external provider-reference file

### DISCUSSION THREAD

**Discussion by olderthandirt5** [CMSgov/price-transparency-guide#443] [General]
One of our partners has said that our provider-reference file is invalid because it contains multiple groups. From everything I have read and heard on the webinars, the point of the external provider-reference file is to have multiple groups. Am I way off base?

I have attached our file which has been validated by the CMS JSON Validator tool. Thanks again for that!!
[2_MRF_provider-reference.txt](https://github.com/CMSgov/price-transparency-guide/files/8476615/2_MRF_provider-reference.txt)

**Comment by shaselton-usds** [comment]
@olderthandirt5 That file contains invalid JSON. Try copying and pasting the contents into an online validator to show that the file contains multiple root nodes which isn't valid.

A valid provider reference file will have a single `provider_groups` array containing all of the possible provider/ein combinations for that network.

**Comment by olderthandirt5** [comment]
I have fixed the JSON file and it has been validated by the Validator tool:
[1validation.txt](https://github.com/CMSgov/price-transparency-guide/files/8481932/1validation.txt)
[provider-reference_1_cms.txt](https://github.com/CMSgov/price-transparency-guide/files/8481942/provider-reference_1_cms.txt)

This is the question I have now:
We have one partner with 287 different provider groups in a single network. In addition to the Table of Contents and the In Network files, do we need to include 287 different provider-reference files? Or is there a way to combine those 287 different provider groups into a single file with different id numbers?

**Comment by olderthandirt5** [comment]
@shaselton-usds Any direction on the question above?

> We have one partner with 287 different provider groups in a single network. In addition to the Table of Contents and the In Network files, do we need to include 287 different provider-reference files? Or is there a way to combine those 287 different provider groups into a single file with different id numbers?

**Reply by shaselton-usds** [reply]
If the 287 different provider groups are in a single network, include them in the same file -- it would be [287 objects](https://github.com/CMSgov/price-transparency-guide/blob/master/examples/provider-reference/provider-reference.json#L3-L13) in the `provider_groups` array. This is assuming that the negotiations of the groups are the same across the entire network. 

The external provider group references aren't included in the table of contents; they are included in the [actual in-network files](https://github.com/CMSgov/price-transparency-guide/blob/master/examples/in-network-rates/in-network-rates-fee-for-service-single-plan-sample.json#L25-L28).

**Comment by olderthandirt5** [comment]
For demonstration purposes, I created a file with only 3 provider groups and ran it through the validator tool. This is how we will generate the provider-reference files and will reference them in the in-network file.
The validation output:

> PS C:\cms\price-transparency-guide-validator> cms-mrf-validator validate c:\cms\data\April\2_MRF_provider-reference.json v1.0.0 -t provider-reference
> Input JSON is valid.
> 
> PS C:\cms\price-transparency-guide-validator> cms-mrf-validator validate c:\cms\data\April\2_MRF_provider-reference.json v1.0.0 -t provider-reference -o c:\cms\414_2_MRF.txt

The validation output file:
[414_2_MRF.txt](https://github.com/CMSgov/price-transparency-guide/files/8493298/414_2_MRF.txt)

The provider-reference file (Converted to text file):
[2_MRF_provider-reference.txt](https://github.com/CMSgov/price-transparency-guide/files/8493307/2_MRF_provider-reference.txt)

I appreciate your help. I will post this file to the "Show and Tell" section later, in case anyone else gets confused. I know I am seeing JSON in my sleep these days.

**Comment by olderthandirt5** [comment]
@shaselton-usds
The attached provider-reference file passes the JSON Validator tool, formats correctly in Notepad++, but running it through an online validation tool results in a condensed version with the provider group #1 only showing the last npi/tin combination

Result of the online JSON validator at DuckDuckGo:
`{
  "provider_groups": [
    {
      "id": 1,
      "npi": [
        1306950175,
        1821191966,
        1174591911,
        1023308780
      ],
      "tin": {
        "type": "ein",
        "value": "470781933"
      }
    },
    {
      "id": 2,
      "npi": [
        1720332489
      ],
      "tin": {
        "type": "ein",
        "value": "611352417"
      }
    },
    {
      "id": 3,
      "npi": [
        1245296664,
        1578768339,
        1770685604,
        1790994960
      ],
      "tin": {
        "type": "ein",
        "value": "426004813"
      }
    }
  ]
}`
[2_MRF_provider-reference0503.txt](https://github.com/CMSgov/price-transparency-guide/files/8611300/2_MRF_provider-reference0503.txt)

Can a provider group have multiple npi/tin combinations?

Thanks again for all your work.

**Comment by skochin** [comment]
@olderthandirt5 The schema allows for one or more objects under the "provider_groups" array, however, each object can only have one "npi" and "tin" property, so I beleive that is why you're seeing the validator only recognize one "npi" and "tin".  Also, there is no defined "id" property in this schema.  To my understanding, the provider-reference schema can only represent one grouping of providers (with "grouping" being any combination of providers and/or groups that share the same rates and therefore can be referred to by one provider_references value in the in-network-rates file).

To include multiple npi/tin combos within one grouping, you would strucuture it as:
{
  "provider_groups": [
    {
      "npi": [
        1111111111,
        2222222222
      ],
      "tin": {
        "type": "ein",
        "value": "123456789"
      }
    },
    {
      "npi": [
        3333333333
      ],
      "tin": {
        "type": "ein",
        "value": "987654321"
      }
    }
  ]
}
